### ISSUE TITLE
InvalidTypeException: Invalid type for value..

### ISSUE THREAD

**Issue by 13o6** [brianfrankcooper/YCSB#395]
I'm currently trying to run a workload on `cassandra-2.1` after I've clone and built the latest ycsb repository. When I try to run the following command: 
<pre>./bin/ycsb load cassandra-cql -p hosts="10.240.73.108" -P workloads/workloada</pre>
I get the following exception:
<pre>com.datastax.driver.core.exceptions.InvalidTypeException: Invalid type for value 1 of CQL type varchar, expecting class java.lang.String but class java.nio.HeapByteBuffer provided
	at com.datastax.driver.core.BoundStatement.bind(BoundStatement.java:190)
	at com.datastax.driver.core.DefaultPreparedStatement.bind(DefaultPreparedStatement.java:103)
	at com.yahoo.ycsb.db.CassandraCQLClient.insert(CassandraCQLClient.java:437)
	at com.yahoo.ycsb.DBWrapper$5.go(DBWrapper.java:281)
	at com.yahoo.ycsb.DBWrapper.operation(DBWrapper.java:289)
	at com.yahoo.ycsb.DBWrapper.insert(DBWrapper.java:268)
	at com.yahoo.ycsb.workloads.CoreWorkload.doInsert(CoreWorkload.java:459)
	at com.yahoo.ycsb.ClientThread$2.doOperation(Client.java:366)
	at com.yahoo.ycsb.ClientThread.run(Client.java:416)
	at com.yahoo.ycsb.ClientThread.run(Client.java:363)
</pre>

My ycsb keyspace has the following schema: 
<pre>cqlsh> DESCRIBE ycsb.usertable 

CREATE TABLE ycsb.usertable (
    y_id text PRIMARY KEY,
    field0 text,
    field1 text,
    field2 text,
    field3 text,
    field4 text,
    field5 text,
    field6 text,
    field7 text,
    field8 text,
    field9 text
) </pre>

Any hints are appreciated

**Comment by busbey** [comment]
AFAIK, we still don't work with Cassandra 2.1+ (see #293, #369).

If you have the time to try to chase down what's not working, we'd be happy to help give pointers.

**Comment by 13o6** [comment]
Thanks for your reply @busbey , I decided to remove the folder and build it again but this time I can't even build it due to the following `BUILD ERROR`:
```bash
org.apache.maven.lifecycle.LifecycleExecutionException: Error reading assemblies: Descriptor with ID 'datastore-specific-assembly' not found
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:719)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalWithLifecycle(DefaultLifecycleExecutor.java:556)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:535)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:387)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:348)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:180)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:328)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:138)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:362)
	at org.apache.maven.cli.compat.CompatibleMain.main(CompatibleMain.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
	at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
	at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error reading assemblies: Descriptor with ID 'datastore-specific-assembly' not found
	at org.apache.maven.plugin.assembly.mojos.AbstractAssemblyMojo.execute(AbstractAssemblyMojo.java:453)
	at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:490)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:694)
	... 17 more
Caused by: org.apache.maven.plugin.assembly.io.AssemblyReadException: Descriptor with ID 'datastore-specific-assembly' not found
	at org.apache.maven.plugin.assembly.io.DefaultAssemblyReader.addAssemblyForDescriptorReference(DefaultAssemblyReader.java:213)
	at org.apache.maven.plugin.assembly.io.DefaultAssemblyReader.readAssemblies(DefaultAssemblyReader.java:133)
	at org.apache.maven.plugin.assembly.mojos.AbstractAssemblyMojo.execute(AbstractAssemblyMojo.java:449)
	... 19 more
```
Not sure whether I should open another issue just for it or it's still relevant here.

**Comment by busbey** [comment]
what folder did you remove? that looks like your working directory has been corrupted the 'datastore-specific-assembly' is a part of the code that ships in YCSB.

**Comment by cmccoy** [comment]
The Cassandra 2.1 issue should be fixed by #454 - closing. Could you re-open if you're still having trouble with the latest version from `master`?
